cypress/integration/alerts_dashboard_flyout_spec.js

```diff
@@ -0,0 +1,287 @@
+  });
+
+  it('Query-level monitor flyout test', () => {
+    // Click the link for the flyout.
+    cy.get(`[data-test-subj="euiLink_${QUERY_TRIGGER}"]`).click();
+
+    // Wait for the flyout to load the trigger-specific alerts.
+    cy.wait(10000);
+
+    // Perform the test checks within the flyout component.
+    cy.get(`[data-test-subj="alertsDashboardFlyout_${QUERY_TRIGGER}"]`).within(() => {
+      // Confirm flyout header contains expected text.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_header_${QUERY_TRIGGER}"]`).contains(
+        `Alerts by ${QUERY_TRIGGER}`
+      );
+
+      // Confirm 'Trigger name' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_triggerName_${QUERY_TRIGGER}"]`).contains(
+        'Trigger name'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_triggerName_${QUERY_TRIGGER}"]`).contains(
+        QUERY_TRIGGER
+      );
+
+      // Confirm 'Severity' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_severity_${QUERY_TRIGGER}"]`).contains(
+        'Severity'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_severity_${QUERY_TRIGGER}"]`).contains(
+        '2 (High)'
+      );
+
+      // Confirm 'Monitor' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_monitor_${QUERY_TRIGGER}"]`).contains(
+        'Monitor'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_monitor_${QUERY_TRIGGER}"]`).contains(
+        QUERY_MONITOR
+      );
+
+      // Confirm 'Conditions' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_conditions_${QUERY_TRIGGER}"]`).contains(
+        'Condition'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_conditions_${QUERY_TRIGGER}"]`).contains(
+        `ctx.results[0].hits.total.value < 10000`
+      );
+
+      // Confirm 'Time range for the last' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_timeRange_${QUERY_TRIGGER}"]`).contains(
+        'Time range for the last'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_timeRange_${QUERY_TRIGGER}"]`).contains(
+        '10 day(s)'
+      );
+
+      // Confirm 'Filters' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_filters_${QUERY_TRIGGER}"]`).contains(
+        'Filters'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_filters_${QUERY_TRIGGER}"]`).contains('-');
+
+      // Confirm 'Group by' sections renders as expected.
+      cy.get(`[data-test-subj="alertsDashboardFlyout_groupBy_${QUERY_TRIGGER}"]`).contains(
+        'Group by'
+      );
+      cy.get(`[data-test-subj="alertsDashboardFlyout_groupBy_${QUERY_TRIGGER}"]`).contains('user');
+
+      // Set the 'severity' filter to only display ACTIVE alerts.
+      cy.get('[data-test-subj="dashboardAlertStateFilter"]').select('Active');
+      cy.wait(10000);
```

Comment:
This `wait()` can be removed. The alerts should have been loaded when the flyout is open.

Reply:
Removed this `wait` period.